### PR TITLE
Routes_opts_improvement

### PR DESCRIPTION
**Pull request by dnlup**
## 🚨 Proposed changes

> Please review the [guidelines for contributing](../../CONTRIBUTING.md) to this repository.

Add the possibility to pass `true` or `false` to toggle all the routes metrics.

## ⚙️ Types of changes

What types of changes does your code introduce? _Put an `x` in the boxes that apply_

-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)
-   [ ] Refactor

### CODE REVIEW

File: tests/metrics-collection.test.js [hunk 1]
```diff
@@ -149,6 +149,91 @@ tap.test('disabling process health metrics', async (t) => {
...
+            namespace: 'enable_routes_test',
+        },
+        routes: true,
+        health: false,
+    });
+    t.teardown(async () => {
+        t.context.statsd.removeAllListeners('metric');
+        return server.close();
+    });
+    await Promise.all([
```
Comment (antoniomuso) [inline tests/metrics-collection.test.js:214]: Here you can use `checkMetrics` function

File: index.js [hunk 2]
```diff
@@ -78,24 +78,48 @@ const DEFAULT_CONFIG_OPTIONS = {
...
+            ...config,
+        };
+    }
+}
+
 module.exports = fp(
     async function (fastify, opts) {
         const { client = {}, routes, ...others } = opts;
-        // TODO: allow to pass routes: false to disable the routes alltoghether.
+        if (opts.routes) {
```
Comment (antoniomuso) [inline index.js:102]: Why don't you use only one `if`? as:
```js
if (opts.routes &&
    typeof opts.routes !== 'boolean' &&
    typeof opts.routes !== 'object') {
          throw new Error('"routes" must be a boolean or a config object.');
}
```

Comment (dnlup) [reply index.js:102]: I though it would have been simpler to read, I'll change it 😉

Comment (antoniomuso) [reply index.js]: No problem, if you think it is more readable you can maintain it. I have the opposite opinion, more indentations there are more unreadable is.

**Review by antoniomuso — APPROVED** [review]
LGTM, GJ :rocket:

**Comment by simonecorsi** [comment]
There are some conflict to resolve! :)

**Comment by dnlup** [comment]
> There are some conflict to resolve! :)

Should be ok now 😉

**Review by simonecorsi — APPROVED** [review]

**Pull request merged**